registry/registry.go

```diff
@@ -737,24 +811,299 @@ func downloadVersion(opts *VersionOptions) (ver *Version, attachments []*kivik.A
 		} else if tryCount <= 3 {
 			continue
 		} else {
-			return
+			return nil, nil, err
 		}
 	}
 
+	// Reader for filesize
 	counter := &Counter{}
 	var reader io.Reader = buf
 	reader = io.TeeReader(reader, counter)
 
-	var packVersion string
+	// Reader for tarball content
+	var fileContent = new(bytes.Buffer)
+	reader = io.TeeReader(reader, fileContent)
+
+	// Reading the tarball content
+	tarball, err := ReadTarballVersion(reader, contentType, url)
+	if err != nil {
+		return nil, nil, err
+	}
+	tarball.ContentType = contentType
+
+	return tarball, counter, nil
+}
+
+func downloadVersion(opts *VersionOptions) (ver *Version, attachments []*kivik.Attachment, err error) {
+	var tarPrefix string
+	url := opts.URL
+
+	tarball, counter, errd := downloadTarball(opts, url)
+	if err != nil {
+		err = multierror.Append(err, errd)
+	}
+	// Retreiving the tarball manifest
+	parsedManifest := tarball.Manifest
+
+	if !tarball.HasPrefix {
+		tarPrefix = ""
+	}
+
+	// Checks
+	if _, erre := tarball.CheckEditor(); erre != nil {
+		err = multierror.Append(err, erre)
+	}
+	if _, errs := tarball.CheckSlug(); errs != nil {
+		err = multierror.Append(err, errs)
+	}
+	if _, errv := tarball.CheckVersion(opts.Version); errv != nil {
+		err = multierror.Append(err, errv)
+	}
+
+	fileContent := new(bytes.Buffer)
+	if _, errc := io.Copy(fileContent, tarball.Content); errc != nil {
```

Comment:
What is the purpose of copying `tarball.Content` to `fileContent`? Is it not possible to use `tarball.Content` below instead of `fileContent`?